src/mbgl/tile/geometry_tile.cpp

```diff
@@ -255,13 +255,37 @@ void GeometryTile::getGlyphs(GlyphDependencies glyphDependencies) {
 }
 
 void GeometryTile::onImagesAvailable(ImageMap images, ImageMap patterns, ImageVersionMap versionMap, uint64_t imageCorrelationID) {
-    worker.self().invoke(&GeometryTileWorker::onImagesAvailable, std::move(images), std::move(patterns), std::move(versionMap), imageCorrelationID);
+    if (!missingImagesRequested) {
+        worker.self().invoke(&GeometryTileWorker::onImagesAvailable,
+                             std::move(images),
+                             std::move(patterns),
+                             std::move(versionMap),
+                             imageCorrelationID);
+    } else {
+        layoutResult->iconAtlas = makeImageAtlas(images, patterns, versionMap);
+        missingImagesRequested = false;
+    }
```

Comment:
This needs a unit test, if for some reason `missingImagesRequested` is stale and set to true, we will never trigger re-layout when image manager provides requested images.